pkg/daemon/ceph/osd/daemon.go

```diff
@@ -601,3 +601,36 @@ func getVolumeGroupName(lvPath string) string {
 
 	return vgSlice[2]
 }
+
+// GetOSDInfoByID returns the OSDInfo using the ceph volume list
+func GetOSDInfoByID(context *clusterd.Context, clusterInfo *client.ClusterInfo, osdID int) (oposd.OSDInfo, error) {
+	// LVM based OSDs
+	osdLVMList, err := GetCephVolumeLVMOSDs(context, clusterInfo, clusterInfo.FSID, "", false, false)
+	if err != nil {
+		return oposd.OSDInfo{}, fmt.Errorf("failed to list lvm osd(s). %v", err)
+	} else {
+		for _, osdInfo := range osdLVMList {
+			if osdInfo.ID == osdID {
+				// Start the sanitizing sequence
+				logger.Infof("starting sanitizating of lvm OSD disk %d", osdID)
+				return osdInfo, nil
+			}
+		}
+	}
+
+	// Raw based OSDs
+	osdRawList, err := GetCephVolumeRawOSDs(context, clusterInfo, clusterInfo.FSID, "", "", "", false, true)
+	if err != nil {
+		logger.Errorf("failed to list raw osd(s). %v", err)
```

Comment:
return the error here instead of just log it and continue?

Reply:
done